xchange-poloniex/src/main/java/info/bitrich/xchangestream/poloniex/PoloniexStreamingMarketDataService.java

```diff
@@ -90,6 +91,12 @@ public PoloniexStreamingMarketDataService(WampStreamingService streamingService)
 
     @Override
     public Observable<Ticker> getTicker(CurrencyPair currencyPair, Object... args) {
```

Comment:
I've been thinking about this approach and how it combines with the principle that a good API should be a minimal one and not repeat itself unnecessarily.

Here's my thinking: the issue is not that the API doesn't provide what we want here.  As you have shown, you can achieve the same thing at the API level using:
```
Observable<Ticker> allTickers = Observable.fromIterable(List.of(tickers))
            .flatMap(pair -> exchange.getStreamingMarketDataService().getTicker(pair));
```
This is a one-liner, so the API is perfectly sufficient. The problem is actually that _the way the exchanges are supporting the existing API_ is inefficient.  This is endemic throughout xchange-stream.

If the implementation of the Poloniex `StreamingMarketDataService` only ever opened a single `1002` channel, `share()`ed it, and _re-used_ that existing `Observable` for any calls to `getTicker(...)`, with slightly different filters for each call but sharing the same underlying stream, then your approach and the above approach would be computationally equivalent, so there would be no performance penalty.

All of which is a long way to say: how about we just fix this obvious performance hole and avoid expanding the API?

Reply:
I follow! Let’s do that instead. I’ll Tag you again when it’s done.